### PR TITLE
feat(Dropdown): add E2E tests

### PR DESCRIPTION
**Pull request by saurabhdaware**
- Add basic E2E tests for Dropdown
- Remove unit tests which were e2e tests in disguise and move them to real e2e
- Fixes a bug that was found while writing these e2e tests 🙈 . The dropdown was getting closed even when the state change on consumer end was not called inside `onOpenChange`
- Fixes onClick getting called twice on Dropdown triggers

### CODE REVIEW

File: packages/blade/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -99,24 +98,22 @@ const _Dropdown = ({
 
   const dropdownBaseId = useId('dropdown');
 
-  useDidUpdate(() => {
-    onOpenChange?.(isOpen);
-    if (!isOpen) {
```
Comment (saurabhdaware) [inline packages/blade/src/components/Dropdown/Dropdown.tsx:104]: With this logic, we were calling `onOpenChange` as well as we were actually closing and opening dropdown.

Ideally we should only call `onOpenChange` wherever state needs to change and then consumers add 

```jsx
<Dropdown isOpen={isDropdownOpen} onOpenChange={(isOpen) => { setIsDropdownOpen(isOpen) }} />
```

if someone doesn't call state change in onOpenChange, then it shouldn't close the dropdown.

File: packages/blade/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -99,24 +98,22 @@ const _Dropdown = ({
...
+  const setIsOpenControlled = (isControlledStateOpen: boolean): void => {
+    onOpenChange?.(isControlledStateOpen);
+    if (!isControlledStateOpen) {
```
Comment (anuraghazra) [inline packages/blade/src/components/Dropdown/Dropdown.tsx:103]: Can we use the useControllableState hook to manage the controlled/uncontrolled behaviour of `isOpen`?

**Review by anuraghazra — APPROVED** [review]
LGTM. 

- Check the tests on Safari/Firefox

File: packages/blade/src/components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Dropdown/Dropdown.tsx:103]: ouu didn't know about this hook. let me see.

Comment (saurabhdaware) [reply packages/blade/src/components/Dropdown/Dropdown.tsx]: done

**Comment by saurabhdaware** [comment]
> LGTM.
>
> Check the tests on Safari/Firefox

done

**Review by anuraghazra — APPROVED** [review]
lint failing. else lgtm

**Review by chaitanyadeorukhkar — APPROVED** [review]

**Pull request merged**